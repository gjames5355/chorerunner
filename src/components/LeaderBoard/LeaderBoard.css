body {
  font-size: 1em;
}

.leaderboard_container {
  background: rgba(0, 0, 0, 0.7);
  /* border-radius: 10px; */
  min-width: 40%;
  margin: 20px auto;
  width: 95%;
  border-radius: 10px;
}

.leader_board {
  font-family: 'Press Start 2P', cursive;
  color: white;
  text-align: center;
  padding: 10px 0;
}

.leader_board h3 {
  color: rgb(255, 217, 0);
}

.headerRow {
  width: auto;
  display: flex;
  padding: 5px 0;
  border-bottom: 1px solid white;
}

.header_rank,
.header_name,
.header_score {
  text-align: center;
}

.score_display {
  width: auto;
  margin: 10px auto;
}

.leader_board-grid {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  font-size: 15px;
}

.rankings {
  /* width: 80%; */
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
  list-style: none;
  margin: 0 auto;
  font-size: 1.1em;
  font-family: 'Electrolize', sans-serif;

}

.rankings li {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}

.rankings li:nth-child(-n + 3):before {
  content: '';
  display: block;
  height: 25px;
  width: 25px;
  background-size: contain;
  background-repeat: no-repeat;
  background-image: url('../../ImgAssets/ChoreRunner.png');
}

.rankings li:nth-child(1):before {
  background-image: url('../../ImgAssets/FirstCup.png');
  display: block;
  margin: 0 auto;
  background-size: 80px;
  height: 100px;
  width: 80px;
}

.rankings li:nth-child(2):before {
  background-image: url('../../ImgAssets/SecondCup.png');
  display: block;
  margin: 0 auto;
  background-size: 80px;
  height: 100px;
  width: 80px;
}

.rankings li:nth-child(3):before {
  background-image: url('../../ImgAssets/ThirdCup.png');
  display: block;
  margin: 0 auto;
  background-size: 80px;
  height: 100px;
  width: 80px;
}

.rankings li:nth-child(-n + 3) .score {
  display: block;
}

.rank {
  font-size: 30px;
}

.name_col {
  text-align: center;
  margin: 0 auto;
  word-wrap: break-word;
}

.score_col {
  text-align: center;
}


@media (min-width:500px) {
  .leaderboard_container {
    width: 70%;
  }

  .rankings {
    font-size: 1.3em;
    font-family: 'Electrolize', sans-serif;
  }
}

@media (min-width:850px) {
  .leaderboard_container {
    width: 700px;
  }

  .rankings {
    font-size: 1.4em;
    font-family: 'Electrolize', sans-serif;
  }
}