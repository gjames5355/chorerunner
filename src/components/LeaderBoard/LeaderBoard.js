import React, { Component } from 'react';
import HouseholdContext from '../../contexts/HouseHoldContext';
import ApiService from '../../services/api-service.js';
import './LeaderBoard.css';

export default class Leaderboard extends Component {
  static contextType = HouseholdContext;

  state = {
    members: [],
  };

  componentDidMount() {
    ApiService.getMemberScores()
      .then(res => {
        this.setState({ members: res });
      })
      .catch(error => this.context.setError(error));
  }

  renderScores() {
    let members = this.state.members;
    return members.map((member, index) => {
      return (
        <li key={member.id} className='score_display'>
          {index > 2 ? <div className='rank'>{`${index + 1}`}</div> : null}
          <div className='name_col'>
            <span>{member.name}</span>
          </div>
          <div className='score_col'>
            <span className='score'>{member.total_score}</span>
          </div>
        </li>
      );
    });
  }

  render() {
    return (
      <section className='leader_board'>
        <h3>Leaderboard</h3>
        <div className='leader_board-grid'>
          <div className='header_rank'>
            <span>Rank</span>
          </div>
          <div className='header_name'>
            <span>Member Name</span>
          </div>
          <div className='header_score'>
            <span>Score</span>
          </div>
        </div>
        <ul className='rankings'>{this.renderScores()}</ul>
      </section>
    );
  }
}
