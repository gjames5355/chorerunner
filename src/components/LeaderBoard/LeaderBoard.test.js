import React from 'react';
import ReactDOM from 'react-dom';
import LeaderBoard from './LeaderBoard';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('LeaderBoard component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <LeaderBoard />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LeaderBoard />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
