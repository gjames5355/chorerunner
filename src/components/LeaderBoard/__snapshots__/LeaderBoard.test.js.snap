// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoard component testing renders UI as expected 1`] = `
<section
  className="leader_board"
>
  <h3>
    Leaderboard
  </h3>
  <div
    className="leader_board-grid"
  >
    <div
      className="header_rank"
    >
      <span>
        Rank
      </span>
    </div>
    <div
      className="header_name"
    >
      <span>
        Member Name
      </span>
    </div>
    <div
      className="header_score"
    >
      <span>
        Score
      </span>
    </div>
  </div>
  <ul
    className="rankings"
  />
</section>
`;
